### PR TITLE
Items

### PR DESCRIPTION
**Pull request by L3nn0x**
create items, drop items, crete zulies, drop zulies, equip items, unequip items

---
This change is [Reviewable](https://reviewable.io/reviews/dev-osrose/osirose-new/177)

### CODE REVIEW

**Review by RavenX8 — DISMISSED** [review]
> Reviewed 19 of 27 files at r1, 9 of 9 files at r2.
> *[Reviewable](https://reviewable.io/reviews/dev-osrose/osirose-new/177)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @rminderhoud)

**Review by RavenX8 — CHANGES_REQUESTED** [review]
> Reviewed 1 of 2 files at r3.
> *[Reviewable](https://reviewable.io/reviews/dev-osrose/osirose-new/177)* status: 27 of 28 files reviewed, 1 unresolved discussion (waiting on @L3nn0x, @RavenX8, and @rminderhoud)

___
*[src/map/src/items/inventory.cpp, line 26 at r3](https://reviewable.io/reviews/dev-osrose/osirose-new/177#-Ld4E0F4BewKOF8LRmy3:-Ld4E0F4BewKOF8LRmy4:b-bzstcz) ([raw file](https://github.com/dev-osrose/osirose-new/blob/1240b3ef018543d701d24a3b68a1e480c7330c74/src/map/src/items/inventory.cpp#L26)):*
> ```cpp
> inline bool is_spot_correct(const EntitySystem& entitySystem, RoseCommon::Entity entity, size_t spot) {
>     const auto& item = entitySystem.get_component<ItemDef>(entity);
>     if (spot > 9) { // non equip items TODO: put that magic value somewhere else
> ```

this doesn't seem right.

**Review by RavenX8 — COMMENTED** [review]
> Reviewed 1 of 2 files at r3.
> *[Reviewable](https://reviewable.io/reviews/dev-osrose/osirose-new/177)* status: all files reviewed, 1 unresolved discussion (waiting on @L3nn0x and @rminderhoud)

**Review by RavenX8 — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/dev-osrose/osirose-new/177)* status: all files reviewed, 1 unresolved discussion (waiting on @L3nn0x and @rminderhoud)

___
*[src/map/src/items/inventory.cpp, line 26 at r3](https://reviewable.io/reviews/dev-osrose/osirose-new/177#-Ld4E0F4BewKOF8LRmy3:-Ld4EbTv4TRJ3h_vzMFb:b-qe8pdm) ([raw file](https://github.com/dev-osrose/osirose-new/blob/1240b3ef018543d701d24a3b68a1e480c7330c74/src/map/src/items/inventory.cpp#L26)):*
<details><summary><i>Previously, RavenX8 (Christopher Torres) wrote…</i></summary><blockquote>

this doesn't seem right.
</blockquote></details>

Item type 14 also needs to be able to be equipped.

File: src/rosecommon/src/itemdb.cpp [hunk 1]
```diff
@@ -33,6 +34,7 @@ void ItemDatabase::initialize() {
       newItem.buy_price = static_cast<uint32_t>(row.priceBuy);
       newItem.sell_price = static_cast<uint32_t>(row.priceSell);
       newItem.weight = static_cast<uint16_t>(row.weight);
+      newItem.is_stackable = newItem.type >= 10 && newItem.type <= 13;
```
Comment (rminderhoud) [inline src/rosecommon/src/itemdb.cpp]: Magic numbers

Comment (RavenX8) [reply src/rosecommon/src/itemdb.cpp]: MAGIC!

**Review by RavenX8 — APPROVED** [review]
> Reviewed 4 of 4 files at r5.
> *[Reviewable](https://reviewable.io/reviews/dev-osrose/osirose-new/177)* status: all files reviewed, 1 unresolved discussion (waiting on @L3nn0x and @rminderhoud)

**Pull request merged**